src/vehicle.cpp

```diff
@@ -2402,6 +2413,82 @@ void Vehicle::HandleLoading(bool mode)
 	this->IncrementImplicitOrderIndex();
 }
 
+/**
+ * Check if the current vehicle has an unbunching order.
+ * @return true Iff this vehicle has an unbunching order.
+ */
+bool Vehicle::HasUnbunchingOrder() const
+{
+	for (Order *o : this->Orders()) {
+		if (o->IsType(OT_GOTO_DEPOT) && o->GetDepotActionType() & ODATFB_UNBUNCH) return true;
+	}
+	return false;
+}
+
+/**
+ * Leave an unbunching depot and calculate the next departure time for shared order vehicles.
+ */
+void Vehicle::LeaveUnbunchingDepot()
+{
+	/* Set the start point for this round trip time. */
+	this->depot_unbunching_last_departure = TimerGameTick::counter;
+
+	/* Tell the timetable we are now "on time." */
+	this->lateness_counter = 0;
+	SetWindowDirty(WC_VEHICLE_TIMETABLE, this->index);
+
+	/* Find the average travel time of vehicles that we share orders with. */
+	uint num_vehicles = 0;
+	TimerGameTick::Ticks total_travel_time = 0;
+
+	Vehicle *u = this->FirstShared();
+	for (; u != nullptr; u = u->NextShared()) {
+		/* Ignore vehicles that are manually stopped or crashed. */
+		if (u->vehstatus & (VS_STOPPED | VS_CRASHED)) continue;
+
+		num_vehicles++;
+		total_travel_time += u->round_trip_time;
+	}
+
+	/* Make sure we cannot divide by 0. */
+	num_vehicles = std::max(num_vehicles, 1u);
+
+	/* We must have a separation of at least one tick. */
+	TimerGameTick::Ticks separation = std::max((total_travel_time / num_vehicles / num_vehicles), 1u);
```

Comment:
I wondered why the total time was divide twice, but I couldn't figure it out myself. So I searched for the original PR, where there is a detailed explanation. So I think it might be worth copying here.
https://github.com/OpenTTD/OpenTTD/blob/bbddd92ec48256133e1de64d1b76e970fd200a36/src/vehicle.cpp#L2429-L2441

Reply:
That explanation is for a different, much more complex algorithm. 😉

It's divided twice because it calculates the average time (`average = total / num_vehicles`) and then determines how much time should be between each vehicle (`separation = average / num_vehicles`).

I'll explain this in the comment.

Reply:
> It's divided twice because it calculates the average time (`average = total / num_vehicles`) and then determines how much time should be between each vehicle (`separation = average / num_vehicles`).
That sentence makes it trivial. I feel like I should have really think it over the first time. :)